packages/grid/x-data-grid-premium/src/hooks/features/aggregation/createAggregationLookup.ts

```diff
@@ -47,6 +47,19 @@ const getAggregationCellValue = ({
     if (aggregationRowsScope === 'filtered' && filteredRowsLookup[rowId] === false) {
       return;
     }
+
+    // If the row is a group, we want to aggregate based on its children
+    // For instance in the following tree, we want the aggregated values of A to be based on A.A, A.B.A and A.B.B but not A.B
```

Comment:
We have it (or at least we could easily change the execution order to have it)
But it would require accessing the value manually instead of using the `getCellValue` method because the value is not yet stored in state.

That could be a future improvement (not sure what the perfect behavior is here).
But I did not want to make deep changes for this bug fix.